transactor/src/main/scala/io/mediachain/transactor/Copycat.scala

```diff
@@ -44,31 +48,116 @@ object Copycat {
       server
     }
   }
+
+  sealed abstract class ClientState
+  
+  object ClientState {
+    case object Connected extends ClientState
+    case object Suspended extends ClientState
+    case object Disconnected extends ClientState
+  }
+
+  trait ClientStateListener {
+    def onStateChange(state: ClientState): Unit
+  }
   
   class Client(client: CopycatClient) extends JournalClient {
+    import scala.concurrent.ExecutionContext.Implicits.global
+    
+    private var shutdown = false
+    private var server: Option[String] = None
     private var listeners: Set[JournalListener] = Set()
+    private var stateListeners: Set[ClientStateListener] = Set()
+    private val logger = LoggerFactory.getLogger(classOf[Client])
+    private val random = new Random
+    private val timer = new Timer(true) // runAsDaemon
+ 
+    client.onStateChange(new Consumer[CopycatClient.State] {
+                              def accept(state: CopycatClient.State) {
+                                onStateChange(state)
+                              }
+    })
     
-    def copycat = client
+    // submit a state machine operation with retry logic to account
+    // for potentially transient client connectivity errors
+    val maxRetries = 5
+    private def submit[T](op: Operation[T], retry: Int = 0): Future[T] = {
+      val fut = FutureConverters.toScala(client.submit(op))
+      if (retry < maxRetries) {
+        fut.recoverWith { 
+          case e: Throwable =>
```

Comment:
fine for now, tho we should def dive into the errors and allow some to halt progress if necessary